cyclonedx/output/json.py

```diff
@@ -151,7 +151,8 @@ def _recurse_specialise_component(self, bom_json: Dict[Any, Any], base_key: str
                 if not self.component_supports_author() and 'author' in bom_json[base_key][i].keys():
                     del bom_json[base_key][i]['author']
 
-                if self.component_version_optional() and bom_json[base_key][i]['version'] == "":
+                if self.component_version_optional() and 'version' in bom_json[base_key][i].keys() and \
```

Comment:
instead of `'version' in bom_json[base_key][i].keys()` 
utilize `bom_json[base_key][i].get('version', '')` with an empty string as default value
?